packages/cadl-ranch-specs/http/type/union/main.tsp

```diff
@@ -79,3 +91,75 @@ This test is testing sending the second union value in named union property.
 @route("/model2")
 @post
 op sendSecondNamedUnionValue(@body input: ModelWithNamedUnionProperty): OkResponse;
+
+@scenario
+@scenarioDoc("""
+This test is tesing receiving a string value in simple union property.
+
+Expect request:
+```json
+{ "kind": "string" }
+```
+
+Expect response:
+```json
+{ "simpleUnion": "string" }
+```
+""")
+@route("/receive/string")
+@post
+op receiveString(@body input: Input): ModelWithSimpleUnionPropertyInResponse;
+
+@scenario
+@scenarioDoc("""
+This test is tesing receiving an int array value in simple union property.
+
+Expect request:
+```json
+{ "kind": "int-array" }
+```
+
+Expect response:
+```json
+{ "simpleUnion": [1, 2] }
+```
+""")
+@route("/receive/int-array")
+@post
```

Comment:
If this is only testing the response could we change `post` to `get` and then remove the request body part?

Reply:
Is there a difference?
Get could have a response, Post could also have a response body.
I could remove the request body, that is fine.
I am fine with changing the verb to get but I wonder if there is a difference

Reply:
I understand both the GET and POST could test the response body but I prefer GET without any request body with following reasons:
- the type info is already in route we don't need to duplicate them in request body
- downstream could focus the existing changing part to do verification in our case that is the response body and no need to care others

I think this is our convertion in cadl-ranch with minimal code to make it work, but feel free to correct me if not!
/cc @weidongxu-microsoft @tadelesh

Reply:
I'd agree with Mary.

My main concern was the request body. Since you are going to remove the request body, and you do not test a POST, a GET is a better solution (IMHO).

There may not be significant difference between POST and GET. But since you are going to modify it anyway, GET is the most simple one when you don't have reason to use other verb.

Reply:
body removed and changed to get